src/iop/watermark.c

```diff
@@ -718,6 +720,38 @@ static gchar *_watermark_get_svgdoc(dt_iop_module_t *self, dt_iop_watermark_data
       g_list_free_full(res, &g_free);
     }
 
+    res = dt_tag_get_list(image->id);
+    gchar *keywords = dt_util_glist_to_str(", ", res);
+    svgdoc = _string_substitute(svgdata, "$(IMAGE.TAGS)", (keywords ? keywords : ""));
+    if(svgdoc != svgdata)
+    {
+      g_free(svgdata);
+      svgdata = svgdoc;
+    }
+    if(keywords)
```

Comment:
No need to check for NULL, `g_free` handles that already.

Reply:
Fixed.